src/renderers/dom/shared/__tests__/ReactDOMServerIntegration-test.js

```diff
@@ -166,4 +185,246 @@ describe('ReactDOMServerIntegration', () => {
       expect(e.firstChild.tagName).toBe('BR');
     });
   });
+  
+  describe('property to attribute mapping', function() {
+    describe('string properties', function() {
+      itRenders('simple numbers', async render => {
+        const e = await render(<div width={30} />);
+        expect(e.getAttribute('width')).toBe('30');
+      });
+
+      itRenders('simple strings', async render => {
+        const e = await render(<div width={'30'} />);
+        expect(e.getAttribute('width')).toBe('30');
+      });
+
+      // this seems like it might mask programmer error, but it's existing behavior.
+      itRenders('string prop with true value', async render => {
+        const e = await render(<a href={true} />);
+        expect(e.getAttribute('href')).toBe('true');
+      });
+
+      // this seems like it might mask programmer error, but it's existing behavior.
+      itRenders('string prop with false value', async render => {
+        const e = await render(<a href={false} />);
+        expect(e.getAttribute('href')).toBe('false');
+      });
+
+      // this seems like somewhat odd behavior, as it isn't how <a html> works
+      // in HTML, but it's existing behavior.
+      itRenders('string prop with true value', async render => {
```

Comment:
(This test is probably unnecessary since it's seen the same by React as `href: true` and I'm sure we'll be careful about compat if we ever change JSX.)